tripy/tripy/flat_ir/flat_ir.py

```diff
@@ -59,14 +59,6 @@ def to_mlir_impl():
                 with ir.InsertionPoint(module.body) as ip:
                     # Lets assume only one function with inline code (#9 will fix it)
                     inp_types = [inp.to_mlir() for inp in self.inputs]
-                    if self.shapes:
-                        assert len(inp_types) == len(self.shapes), "mismatched number of inputs vs number of shapes"
-                        for i, (inp_type, inp_shape) in enumerate(zip(inp_types, self.shapes)):
-                            if inp_shape.has_static_shape():
-                                inp_types[i] = ir.RankedTensorType.get(
```

Comment:
Why do we do not need this anymore? Is it because static shape folding is now handled by individual operations?

Reply:
I implemented this via https://github.com/NVIDIA/TensorRT-Incubator/pull/43